### PR TITLE
`Lab. 1`, `Lab. 2` _and_ `Lab. 3` is done.

### PR DESCRIPTION
**Pull request by Egorkin-enabled**
**Hello**, Eugeny Sergeevich. I'm Kozlov Egor, student of 2/278 group.
I want to say that I have written all **CodeWars** tasks. Also made **TS** laboratory with **Unit tests**.

At **CodeWars** I've used different languages:

- **C#** as the main one;
- **JS** for some tasks that couldn't be solved different;
- **C** for about of 1/3 tasks &mdash; I'm pointer arithmetic believer!
- **C++** in few tasks where **C** was forbidden. 

Is everything did OK? I'm waiting for a feedback. Thank you and have a nice day!

### CODE REVIEW

**Comment by Egorkin-enabled** [comment]
Hello Eugenii Sergeevich. I have merged your fixes.

**Comment by Egorkin-enabled** [comment]
Hello again. I have did `Lab 3`. I'll fix `Pull request` header.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (jskonst) [inline package.json:1]: этот тоже удаляем

File: rpgsaga/saga/.gitignore [hunk 2]
```diff
@@ -0,0 +1,130 @@
+# Logs
```
Comment (jskonst) [inline rpgsaga/saga/.gitignore:1]: в каталоге выше был gitignore

File: rpgsaga/saga/src/lab1/funcs.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  const itemCount: number = Math.floor((xEnd - xStart) / xDelta) + 1;
+
+  if (itemCount <= 0) {
+    return [];
+  }
+
+  const arr = new Array<number>(itemCount);
+  let xCurrent = xStart;
+  let index = 0;
+  while (xCurrent < xEnd) {
```
Comment (jskonst) [inline rpgsaga/saga/src/lab1/funcs.ts:20]: for ?

File: rpgsaga/saga/src/lab1/funcs.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    index++;
+    xCurrent += xDelta;
+  }
+
+  return arr;
+}
+
+export function calculatePoints(a: number, b: number, xes: Array<number>): Array<number> {
+  const results: Array<number> = new Array<number>(xes.length);
+  for (const index in xes) {
```
Comment (jskonst) [inline rpgsaga/saga/src/lab1/funcs.ts:31]:
```suggestion
  for (const index of xes) {
```

File: rpgsaga/saga/test/lab1/test.spec.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import * as pointEngine from '../../src/lab1/funcs';
+
+// Let test our task.
+it('should be empty if end >= start.', () => {
```
Comment (jskonst) [inline rpgsaga/saga/test/lab1/test.spec.ts:4]: объедините в describe

**Review by jskonst — CHANGES_REQUESTED** [review]

**Comment by Egorkin-enabled** [comment]
Hello, Eugeny Sergeevich. I've applied all of the remarks. I hope it's all right.

File: rpgsaga/saga/src/lab3/animal.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
+const WRONG_COLOR_FORMAT = new Error('Wrong color format');
+
+export abstract class Animal {
+  private _name: string;
+  private _color: string;
+  private _birthDate: Date;
+
+  constructor(name: string, birthdate: Date, color: string) {
+    this._name = name;
```
Comment (jskonst) [inline rpgsaga/saga/src/lab3/animal.ts:9]:
```suggestion
    this.name = name;
```

File: rpgsaga/saga/src/lab3/animal.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    this._birthDate = birthdate;
```
Comment (jskonst) [inline rpgsaga/saga/src/lab3/animal.ts:10]:
```suggestion
    this.birthDate = birthdate;
```

**Review by jskonst — APPROVED** [review]

**Pull request merged**